### PR TITLE
Dynamically allocate Instance

### PR DESCRIPTION
**Pull request by gumb0**
Required for https://github.com/wasmx/fizzy/pull/246

### CODE REVIEW

**Comment by gumb0** [comment]
clang-tidy false positive seems to disappear, so I removed `NOLINT`

**Comment by gumb0** [comment]
clang-tidy warning still happens in #246, so I won't remove `NOLINT` it here.

**Comment by gumb0** [comment]
Rebased.

**Comment by gumb0** [comment]
Rebased.

**Comment by gumb0** [comment]
Rebased.

File: lib/fizzy/execute.hpp [hunk 1]
```diff
@@ -64,10 +64,23 @@ struct Instance
     std::vector<uint64_t> globals;
     std::vector<ExternalFunction> imported_functions;
     std::vector<ExternalGlobal> imported_globals;
+
+    Instance(Module _module, bytes_ptr _memory, size_t _memory_max_pages, table_ptr _table,
```
Comment (axic) [inline lib/fizzy/execute.hpp:68]: `explicit`?

Comment (gumb0) [reply lib/fizzy/execute.hpp:68]: `explicit` is useful only for single parameter constructors (or the ones where parameters after the first one have default values), others can only be called explicitly anyway.

Comment (axic) [reply lib/fizzy/execute.hpp:68]: Ok, C++ noob here.

**Pull request merged**